### ISSUE TITLE
integration test: mTLS for opaque TCP traffic

### ISSUE THREAD

**Issue by olix0r** [linkerd/linkerd2#3515]
Write an integration test that ensures that a client proxy records the proper metrics (i.e. reflecting mutual identity, pod metadata) when a client application makes an opaque TCP connection to a meshed server.

We should test at least:
* Kubernetes Service IPs;
* Individual pod IPs.

**Comment by krzysztofdrys** [comment]
How is this different from the tests already present in `test/integration/deep/opaqueports`? 🤔 Aren't they doing exactly this?

**Comment by mateiidavid** [comment]
Yeah I think you're right @krzysztofdrys, this seems to already be covered. We test that client apps have expected metrics. I'll go ahead and close this.

**Comment by mateiidavid** [comment]
Well, seems that we were wrong :D The opaque ports tests we currently have all check whether an inbound TCP connection has been opened on the server. We can probably do the same for clients to be sure outbound conn metrics are properly recorded.  Going to re-open this.
